### PR TITLE
Add lint test for cleanup white noise

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Xiang Dai <764524258@qq.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -173,6 +173,8 @@ $(EXES): loki-build-image/$(UPTODATE)
 
 lint: loki-build-image/$(UPTODATE)
 	GOGC=20 golangci-lint run
+	@git diff --name-only HEAD master | xargs --no-run-if-empty -n1 sed -i "s#[ \t]*$##"
```
Comment (Kuqd) [inline Makefile]: can you explain what that does ?

Comment (daixiang0) [reply Makefile]: use `sed` to cleanup white noise, and then use `git diff` to check any diffenerces.

**Comment by daixiang0** [comment]
@Kuqd any update?

**Comment by daixiang0** [comment]
@Kuqd any update?

**Comment by Kuqd** [comment]
It is not clear to me why we need this.

**Comment by daixiang0** [comment]
It works as a text lint tool to ensure no white noise.

**Comment by Kuqd** [comment]
I haven't yet feel the need for this, is that causing you trouble on your side ? and how ? Thanks  !

**Comment by daixiang0** [comment]
I think no white noise is good for all repos, also coders would remove white noise in own or others' codes.
With this check, it would be cool.

**Comment by rfratto** [comment]
@daixiang0 Thank you for the pull request, but I'm not sure if we should have this. I feel that linting should be constrained to issues that are clearly visible in a normal text editor. I wouldn't want a contributor to have their PR fail the CI checks just because they had lines with nothing but whitespace in them, something that they would have to go out of their way to find.

**Pull request closed without merging**